### PR TITLE
WT-7783 Fix RTS to restore tombstone when an on-disk update is out of order prepared

### PR DESCRIPTION
**Pull request by kommiharibabu**
When a prepared update is an out of order update, as part of reconciliation
the earlier updates doesn't modified until it gets committed. while aborting these
uncommitted prepared updates by RTS, it missed to restore the tombstone
from the history store as it is greater than the on-disk update.

### CODE REVIEW

File: src/txn/txn_rollback_to_stable.c [hunk 1]
```diff
@@ -508,11 +508,10 @@ __rollback_ondisk_fixup_key(WT_SESSION_IMPL *session, WT_REF *ref, WT_PAGE *page
...
 
             /*
-             * We have a tombstone on the original update chain and it is behind the stable
-             * timestamp, we need to restore that as well.
+             * We have a tombstone on the original update chain and it is behind or equal to the
+             * stable timestamp, we need to restore that as well.
              */
-            if (hs_stop_durable_ts <= rollback_timestamp &&
-              hs_stop_durable_ts < newer_hs_durable_ts) {
+            if (hs_stop_durable_ts <= rollback_timestamp) {
```
Comment (quchenhao) [inline src/txn/txn_rollback_to_stable.c:514]: Can we add some sorts of assert here to check the on disk value is an out of order prepared update and in the other cases hs_stop_durable_ts < newer_hs_durable_ts should be true?

Comment (kommiharibabu) [reply src/txn/txn_rollback_to_stable.c:514]: Yes, the assert is added to verify the restored tombstone timestamp must be less than the newer update timestamp or an out of order prepared update.

**Review by hbokhari — APPROVED** [review]
LGTM

**Comment by hbokhari** [comment]
@kommiharibabu - can you please test these changes with MongoDB patch build as well?

**Comment by kommiharibabu** [comment]
@hbokhari The Stress and patch build test links are added to the ticket.

**Review by quchenhao — APPROVED** [review]
lgtm

**Pull request merged**